common/src/main/java/io/druid/collections/ReferenceCountingResourceHolder.java

```diff
@@ -50,92 +63,105 @@ public ReferenceCountingResourceHolder(final T object, final Closeable closer)
   @Override
   public T get()
   {
-    synchronized (lock) {
-      if (refcount <= 0) {
-        throw new ISE("Already closed!");
-      }
-
-      return object;
+    if (refCount.get() <= 0) {
+      throw new ISE("Already closed!");
     }
+    return object;
   }
 
   public Releaser increment()
   {
-    synchronized (lock) {
-      if (refcount <= 0) {
-        throw new ISE("Already closed!");
-      }
+    if (refCount.getAndIncrement() <= 0) {
+      refCount.decrementAndGet();
```

Comment:
When two or more threads call increment() on already closed holders, it looks that some threads might pass this checking accidently because above two operations are not atomic.